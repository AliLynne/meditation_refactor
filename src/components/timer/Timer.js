import React from 'react'

const Timer = () => {

  const startTimer = () => {
    console.log('Start Timer!')
  }

  const pauseTimer = () => {
    console.log('Pause Timer!')
  }

  const saveData = () => {
    console.log('Save Data!')
  }
  return (
    <div>
      <h3>00:00:00</h3>
      <button className='button-start' onClick={startTimer}>Start</button>
      <button className='button-pause' onClick={pauseTimer}>Pause</button>
      <button className='button-save' onClick={saveData}>Save</button>
    </div>
  )
}

export default Timer
