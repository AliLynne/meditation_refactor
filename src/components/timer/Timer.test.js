import React from 'react'
import { shallow } from 'enzyme'

import Timer from './Timer'

describe('Timer', () => {
  const mockSave = jest.fn()
  const timer = shallow(<Timer />)

  it('should render correctly', () => {


    expect(timer.debug()).toMatchSnapshot()
  })

  describe('when clicking `Save` button', () => {
    beforeEach(() => {
      timer.find('.button-save').simulate('click')
    })
  })


})
