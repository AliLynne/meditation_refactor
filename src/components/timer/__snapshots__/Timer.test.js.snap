// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer should render correctly 1`] = `
"<div>
  <h3>
    00:00:00
  </h3>
  <button className=\\"button-start\\" onClick={[Function: startTimer]}>
    Start
  </button>
  <button className=\\"button-pause\\" onClick={[Function: pauseTimer]}>
    Pause
  </button>
  <button className=\\"button-save\\" onClick={[Function: saveData]}>
    Save
  </button>
</div>"
`;
